### PR TITLE
feat: popup video panels disabler

### PR DESCRIPTION
**Pull request by OxrxL**
https://github.com/revanced/revanced-integrations/pull/145

### CODE REVIEW

File: src/main/kotlin/app/revanced/patches/youtube/layout/playerpopuppanels/patch/PlayerPopupPanelsPatch.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                StringResource("revanced_player_popup_panels_summary_on", "Player popup panels are enabled"),
+                StringResource("revanced_player_popup_panels_summary_off", "Player popup panels are disabled")
+            )
+        )
+
+        val engagementPanelControllerMethod = EngagementPanelControllerFingerprint.result!!.mutableMethod
+
+        engagementPanelControllerMethod.addInstructions(
+            0, """
+            invoke-static { }, Lapp/revanced/integrations/patches/DisablePlayerPopupPanelsPatch;->disablePlayerPopupPanels()Z
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/playerpopuppanels/patch/PlayerPopupPanelsPatch.kt:46]: A PR for this integrations code is missing. Please do before marking this PR as ready.

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

File: src/main/kotlin/app/revanced/patches/youtube/layout/playerpopuppanels/patch/PlayerPopupPanelsPatch.kt (hunk 1 above)
Comment (OxrxL) [reply src/main/kotlin/app/revanced/patches/youtube/layout/playerpopuppanels/patch/PlayerPopupPanelsPatch.kt:46]: Sorry, was ready, but I forget to push it xD. Anyway:

https://github.com/revanced/revanced-integrations/pull/145

File: src/main/kotlin/app/revanced/patches/youtube/layout/playerpopuppanels/patch/PlayerPopupPanelsPatch.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            move-result v0
+            if-eqz v0, :player_popup_panels
+            if-eqz p4, :player_popup_panels
+            const/4 v0, 0x0
+            return-object v0
+            :player_popup_panels
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/playerpopuppanels/patch/PlayerPopupPanelsPatch.kt:52]: In the future, if you need to jump to an instruction outside of the instructions you compile, you can check out the second parameter of `addInstructions`.

**Review by oSumAtrIX — APPROVED** [review]

File: src/main/kotlin/app/revanced/patches/youtube/layout/playerpopuppanels/patch/PlayerPopupPanelsPatch.kt [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import app.revanced.patches.youtube.layout.playerpopuppanels.annotations.PlayerPopupPanelsCompatibility
+import app.revanced.patches.youtube.layout.playerpopuppanels.fingerprints.EngagementPanelControllerFingerprint
+import app.revanced.patches.youtube.misc.integrations.patch.IntegrationsPatch
+import app.revanced.patches.youtube.misc.settings.bytecode.patch.SettingsPatch
+import app.revanced.patches.youtube.misc.settings.framework.components.impl.StringResource
+import app.revanced.patches.youtube.misc.settings.framework.components.impl.SwitchPreference
+
+@Patch
+@DependsOn([IntegrationsPatch::class, SettingsPatch::class])
+@Name("hide-player-popup-panels")
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/playerpopuppanels/patch/PlayerPopupPanelsPatch.kt:22]:
```suggestion
@Name("disable-auto-player-popup-panels")
```

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

**Pull request merged**